library/pkwrite.c

```diff
@@ -451,6 +454,19 @@ int mbedtls_pk_write_key_der( mbedtls_pk_context *key, unsigned char *buf, size_
 #define PRV_DER_MAX_BYTES   RSA_PRV_DER_MAX_BYTES > ECP_PRV_DER_MAX_BYTES ? \
                             RSA_PRV_DER_MAX_BYTES : ECP_PRV_DER_MAX_BYTES
 
+/* 
+ * PKCS8 envelope:
+ * PrivateKeyInfo ::= SEQUENCE {
+ *       version                   Version,
+ *       privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
+ *       privateKey                PrivateKey,
+ *       attributes           [0]  IMPLICIT Attributes OPTIONAL }
+ * }
+ */
+
+#define PRV_PKCS8_DER_MAX_BYTES   22 + PRV_DER_MAX_BYTES
```

Comment:
Could you please document where the +22 comes from? Perhaps annotate the PrivateKeyInfo structure similar to https://github.com/ARMmbed/mbedtls/pull/1759/files#diff-788a5101fe2618b1e14eb09a62e1d7ecR436